src/dc_federated/backend/zmq_interface.py

```diff
@@ -0,0 +1,107 @@
+"""
+Two classes for defining the ZeroMQ interface between the server and the model.
+"""
+
+import zmq
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
+
+
+class ZMQInterfaceModel:
+    def __init__(
+        self,
+        socket,
+        register_worker_callback,
+        unregister_worker_callback,
+        return_global_model_callback,
+        is_global_model_most_recent,
+        receive_worker_update_callback,
+        server_subprocess_args,
+    ) -> None:
+        self.socket = socket
+        self.register_worker_callback = register_worker_callback
+        self.unregister_worker_callback = unregister_worker_callback
+        self.return_global_model_callback = return_global_model_callback
+        self.is_global_model_most_recent = is_global_model_most_recent
+        self.receive_worker_update_callback = receive_worker_update_callback
+        self.server_subprocess_args = server_subprocess_args
+
+    def receive(self):
+        message = self.socket.recv_multipart()
+        logger.debug(f"Zmq message received: {message[0]}")
+
+        # Server initialisation data request
+        if message[0] == b"server_args_request":
+            self.socket.send_pyobj(self.server_subprocess_args)
+        # Federated Learning API
+        elif message[0] == b"register_worker":
+            self.register_worker_callback(message[1].decode("utf-8"))
+            self.socket.send(b"1")
```

Comment:
what is this `b"1"` - just an acknowledgement of receipt?

Reply:
Exactly. Although it's not currently used.
You do have to send something to unlock the next receive call on the socket I believe.

Reply:
I'll write a test to check what happens if the function raises and exception and the socket does not return. May need to make a decision on how we want to handle that exception.